routes/duties.js

```diff
@@ -0,0 +1,139 @@
+import { ObjectId } from 'mongodb';
+import {
+  addNewDuty, lookForAllDuties, lookForDutyById, deleteDutyById, updateDuty,
+} from '../database.js';
+
+const newDutySchema = {
+  schema: {
+    body: {
+      type: 'object',
+      additionalProperties: false,
+      required: ['name', 'location', 'time', 'constraints', 'soldiersRequired', 'value'],
+      properties: {
+        name: { type: 'string' },
+        location: { type: 'string' },
+        time: {
+          type: 'object',
+          properties: {
+            start: { type: 'string', format: 'date' },
+            end: { type: 'string', format: 'date' },
+          },
+        },
+        constraints: { type: 'array' },
+        soldiersRequired: { type: 'number' },
+        value: { type: 'number' },
+      },
+
+    },
+  },
+
+};
+
+const getDutySchema = {
+  schema: {
+    query: {
+      type: 'object',
+      properties: {
+        name: { type: 'string' },
+        location: { type: 'string' },
+        time: {
+          type: 'object',
+          properties: {
+            start: { type: 'string', format: 'date' },
+            end: { type: 'string', format: 'date' },
+          },
+        },
+        constraints: { type: 'array' },
+        soldiersRequired: { type: 'number' },
+        value: { type: 'number' },
+      },
+    },
+  },
+
+};
+
+const dutyEditSchema = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    location: { type: 'string' },
+    time: {
+      type: 'object',
+      properties: {
+        start: { type: 'string', format: 'date' },
+        end: { type: 'string', format: 'date' },
+      },
+    },
+    constraints: { type: 'array' },
+    soldiersRequired: { type: 'number' },
+    value: { type: 'number' },
+  },
+
+};
+
+export default function dutyRouter(app, opts, done) {
+  app.post('/', newDutySchema, async (req, res) => {
+    const dutyInserted = req.body;
+    const dutyCreated = await addNewDuty(dutyInserted);
+    res.status(201).send({ message: `duty created! _id is: ${dutyCreated.insertedId}` });
```

Comment:
this is nice api for humans (in swagger for example) but it will be hard for program to communicate with such a response. you can see in the assignment what i think the response should look like

Reply:
fixed